tests/codegen/llvm/array_integer_equal_comparison.ll

```diff
@@ -32,84 +35,102 @@ entry:
   %9 = load i64, i64* %"$a", align 8
   %10 = load i64, i64* %"$b", align 8
-  %11 = bitcast i32* %ll to i8*
+  %11 = bitcast i32* %v1 to i8*
   call void @llvm.lifetime.start.p0i8(i64 -1, i8* %11)
-  %12 = bitcast i32* %rr to i8*
+  %12 = bitcast i32* %v2 to i8*
   call void @llvm.lifetime.start.p0i8(i64 -1, i8* %12)
   %13 = bitcast i1* %arraycmp.result to i8*
   call void @llvm.lifetime.start.p0i8(i64 -1, i8* %13)
-  store i1 false, i1* %arraycmp.result, align 1
-  %14 = add i64 %9, 0
-  %probe_read_kernel = call i64 inttoptr (i64 113 to i64 (i32*, i32, i64)*)(i32* %ll, i32 4, i64 %14)
-  %15 = load i32, i32* %ll, align 4
-  %16 = add i64 %10, 0
-  %probe_read_kernel2 = call i64 inttoptr (i64 113 to i64 (i32*, i32, i64)*)(i32* %rr, i32 4, i64 %16)
-  %17 = load i32, i32* %rr, align 4
-  %arraycmp.cmp = icmp ne i32 %15, %17
-  br i1 %arraycmp.cmp, label %arraycmp.false, label %arraycmp.loop
+  store i1 true, i1* %arraycmp.result, align 1
+  %14 = inttoptr i64 %9 to [4 x i32]*
+  %15 = inttoptr i64 %10 to [4 x i32]*
+  %16 = bitcast i32* %i to i8*
+  call void @llvm.lifetime.start.p0i8(i64 -1, i8* %16)
+  %17 = bitcast i32* %n to i8*
+  call void @llvm.lifetime.start.p0i8(i64 -1, i8* %17)
+  store i32 0, i32* %i, align 4
+  store i32 4, i32* %n, align 4
+  br label %while_cond
 
-if_body:                                          ; preds = %arraycmp.false
+if_body:                                          ; preds = %arraycmp.done
   %18 = bitcast i64* %perfdata to i8*
   call void @llvm.lifetime.start.p0i8(i64 -1, i8* %18)
   store i64 30000, i64* %perfdata, align 8
   %pseudo = call i64 @llvm.bpf.pseudo(i64 1, i64 0)
-  %perf_event_output = call i64 inttoptr (i64 25 to i64 (i8*, i64, i64, i64*, i64)*)(i8* %0, i64 %pseudo, i64 4294967295, i64* %perfdata, i64 8)
-  %19 = bitcast i64* %perfdata to i8*
-  call void @llvm.lifetime.end.p0i8(i64 -1, i8* %19)
-  ret i64 0
+  %ringbuf_output = call i64 inttoptr (i64 130 to i64 (i64, i64*, i64, i64)*)(i64 %pseudo, i64* %perfdata, i64 8, i64 0)
+  %if_negative = icmp slt i64 %ringbuf_output, 0
+  br i1 %if_negative, label %event_loss_counter, label %counter_merge
 
-if_end:                                           ; preds = %deadcode, %arraycmp.false
+if_end:                                           ; preds = %deadcode, %arraycmp.done
   ret i64 0
 
-arraycmp.false:                                   ; preds = %arraycmp.done, %arraycmp.loop7, %arraycmp.loop3, %arraycmp.loop, %entry
-  %20 = load i1, i1* %arraycmp.result, align 1
-  %21 = bitcast i1* %arraycmp.result to i8*
-  call void @llvm.lifetime.end.p0i8(i64 -1, i8* %21)
-  %22 = bitcast i32* %ll to i8*
-  call void @llvm.lifetime.end.p0i8(i64 -1, i8* %22)
-  %23 = bitcast i32* %rr to i8*
-  call void @llvm.lifetime.end.p0i8(i64 -1, i8* %23)
-  %24 = zext i1 %20 to i64
-  %true_cond = icmp ne i64 %24, 0
-  br i1 %true_cond, label %if_body, label %if_end
+while_cond:                                       ; preds = %arraycmp.loop, %entry
```

Comment:
This looks like it's using the loop version instead of the unrolled one which should be used by this test, shouldn't it?